src/pybullet_helpers/link.py

```diff
@@ -10,34 +10,49 @@
 
 
 class LinkState(NamedTuple):
-    """Link state from Pybullet."""
+    """Link state to match the output of the PyBullet getLinkState API.
+
+    We use a NamedTuple as it supports retrieving by integer indexing
+    and most closely follows the PyBullet API.
+    """
     linkWorldPosition: Pose3D
     linkWorldOrientation: Quaternion
     localInertialFramePosition: Pose3D
     localInertialFrameOrientation: Quaternion
     worldLinkFramePosition: Pose3D
     worldLinkFrameOrientation: Quaternion
 
+    @property
     def com_pose(self) -> Pose:
         """Center of mass (COM) pose of link."""
         return Pose(self.linkWorldPosition, self.linkWorldOrientation)
 
+    @property
     def pose(self) -> Pose:
         """Pose of link in world frame."""
         return Pose(self.worldLinkFramePosition,
                     self.worldLinkFrameOrientation)
 
 
-def get_link_state(body: int, link: int, physics_client_id: int) -> LinkState:
+def get_link_state(body: int,
+                   link: int,
+                   physics_client_id: int,
+                   compute_forward_kinematics: bool = True) -> LinkState:
     """Get the state of a link in a given body."""
-    link_state = p.getLinkState(body, link, physicsClientId=physics_client_id)
+    # Note: does computeForwardKinematics even do anything? Caelan and Rachel
+    # use it with False
```

Comment:
Need to look into this

Reply:
I'm not sure if we ever verified that it does anything, but I believe the reason that we added it was so that we could reset joint states and then get the corresponding end effector poses during motion planning, _without_ calling stepSimulation().

Reply:
So I just tried a little script to test things out, and it seems like it doesn't make any difference.

https://gist.github.com/williamshen-nz/9f287f23486d124ce664ba5b35edcc77

The only way to really tell is to probably dig into the Bullet source code, but I think that's more effort than it's worth.

Given that Caelan and Rachel don't use it and the PyBullet default is False, should we just set the default to False and add a comment that the behavior with computeForwardKinematics=True doesn't seem to change anything?

Reply:
interesting, thanks for testing that! I agree it's not worth digging into Bullet.

why don't we add that comment, but just leave out the keyword argument altogether if it doesn't seem to make a difference?

we should also make a mental note to check this if we see any regression in the large-scale runs that we'll do after merging